packages/frontend-v2/utils/plan/updatePlanOnDragEnd.ts

```diff
@@ -38,23 +47,52 @@ export const updatePlanOnDragEnd = (
       );
     }
 
-    if (!draggedCourse.data.current.isFromSidebar) {
-      // Course is from a term, so we need to move it, we don't need to move
-      // courses that are from the sidebar.
+    const draggedCourseDetails: ScheduleCourse2<unknown> =
+      draggedCourse.data.current.course;
 
-      // remove the class from the old term and add it to the new term
-      const oldTerm = scheduleTerms.find((term) =>
-        term.classes.some((course) => course.id === draggedCourse.id)
+    const oldTerm = scheduleTerms.find((term) =>
+      term.classes.some((course) => course.id === draggedCourse.id)
+    );
+
+    const isFromSidebar = draggedCourse.data.current.isFromSidebar;
+    const isSameTerm = !isFromSidebar && oldTerm && oldTerm.id === newTerm.id;
+
+    /*
+     * Prevent duplicate courses in the same term,
+     * don't need to display error if course is not changing terms.
+     */
+    if (
+      isCourseInTerm(
+        draggedCourseDetails.classId,
+        draggedCourseDetails.subject,
+        newTerm
+      ) &&
+      !isSameTerm
+    ) {
+      toast.error(
+        `Oops, ${getCourseDisplayString(
+          draggedCourseDetails
+        )} already exists in Year ${newTerm.year}, ${getSeasonDisplayWord(
+          newTerm.season
+        )}.`
       );
+      throw new Error("Duplicate course in term.");
+    }
 
+    /*
+     * Course is from a term, so we need to move it, we don't need to move
+     * courses that are from the sidebar.
+     */
+    if (!draggedCourse.data.current.isFromSidebar) {
```

Comment:
This could reuse the isFromSidebar constant from above.

Reply:
my badd, thank you!!